diff --git a/org/geotools/geometry/jts/CompoundCurve.java b/org/geotools/geometry/jts/CompoundCurve.java
index 3f530bd..77dd84a 100644
--- a/org/geotools/geometry/jts/CompoundCurve.java
+++ b/org/geotools/geometry/jts/CompoundCurve.java
@@ -376,4 +376,4 @@ public boolean isRectangle() {
-        return linearized.isRectangle();
+        return linearize().isRectangle();
     }
 
     public Coordinate[] getCoordinates() {
